File: src/givetime.php
<?php

function givetime_cli() {
  $minutes = $_SERVER['argv'][3];
  if ((string)((int)$minutes) !== ((string)$minutes)) {
    $result = array('error' => 'not an integer');
  }
  else {
    $result = givetime_action($_SERVER['argv'][2], $_SERVER['argv'][3]);
  }
  if (isset($result['error'])) {
    print $result['error']."\n";
    exit(1);
  }
}

function givetime_action($pid, $minutes) {
  global $dbh;

  $dbh->beginTransaction();

  $sth = $dbh->prepare('SELECT * FROM users WHERE pid = :pid');
  $sth->execute(array(':pid' => $pid));
  $user = $sth->fetchObject();
  if ($user === false) return array('error' => 'invalid pid');

  $sth = $dbh->prepare('UPDATE users SET begintime = begintime + :time WHERE pid = :pid');
  $sth->execute(array(':pid' => $pid, ':time' => $minutes * 60));

  $dbh->commit();

  return array();
}

File: testovac/givetime.py
# -*- coding: utf-8 -*-

import os
from models import Users, user_closed


def givetime(app, pid, minutes):
    minutes = int(minutes)

    db = app.DbSession()

    user = db.query(Users).filter_by(pid=pid).first()
    if not user: raise ValueError('invalid pid')

    db.execute(Users.update().where(Users.c.pid==pid).
               values(begintime=user.begintime + minutes * 60))

    db.commit()
    db.close()
givetime.help = '  $0 givetime <pid> <extra minutes>'


commands = {
    'givetime': givetime,
}
